ols/src/query_helpers/query_docs.py

```diff
@@ -0,0 +1,118 @@
+            perform_query: Whether RAG retrieval needs to be done or not.
+            vectordb: Vector db from which you need to retrieve the docs.
+            search_type (Optional[str]): Defines the type of search that
+                the retriever should perform.
+                Can be "similarity" (default), "mmr", or
+                "similarity_score_threshold".
+            search_kwargs (Optional[Dict]): Keyword arguments to pass to the
+                search function. Can include things like:
+                    k: Amount of documents to return (Default: 4)
+                    score_threshold: Minimum relevance threshold
+                        for similarity_score_threshold
+                    fetch_k: Amount of documents to pass to MMR algorithm (
+                    Default: 20)
+                    lambda_mult: Diversity of results returned by MMR;
+                        1 for minimum diversity and 0 for maximum. (Default:
+                        0.5)
+                    filter: Filter by document metadata
+
+        Returns:
+            List[Document]: Returns a list of docs.
+
+        Examples:
+           code-block:: python
+
+            # Retrieve more documents with higher diversity
+            # Useful if your dataset has many similar documents
+            docsearch.as_retriever(
+                search_type="mmr",
+                search_kwargs={'k': 6, 'lambda_mult': 0.25}
+            )
+
+            # Fetch more documents for the MMR algorithm to consider
+            # But only return the top 5
+            docsearch.as_retriever(
+                search_type="mmr",
+                search_kwargs={'k': 5, 'fetch_k': 50}
+            )
+
+            # Only retrieve documents that have a relevance score
+            # Above a certain threshold
+            docsearch.as_retriever(
+                search_type="similarity_score_threshold",
+                search_kwargs={'score_threshold': 0.8}
+            )
+
+            # Only get the single most similar document from the dataset
+            docsearch.as_retriever(search_kwargs={'k': 1})
+
+            # Use a filter to only retrieve documents from a specific paper
+            docsearch.as_retriever(
+                search_kwargs={'filter': {'paper_title':'GPT-4 Technical
+                Report'}}
+            )
+        """
+        self.logger.info(
+            f"""Retrieving docs for
+                query: {query},
+                perform_query: {perform_query},
+                vectordb: {vectordb},
+                search_type: {search_type},
+                search_kwargs: {search_kwargs}
+        """
+        )
+
+        if not perform_query:
+            return None
+
+        if search_type not in ("mmr", "similarity", "similarity_score_threshold"):
+            self.logger.error(f"incorrect search type {search_type}")
+            raise RetrieveDocsExceptionError("search type in invalid")
```

Comment:
i'd include the value of search_type that was provided, in the exception.  and i think `in` should be `is`

Reply:
done, will be seen in next push